fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/IndexEntry.java

```diff
@@ -66,29 +74,76 @@ public IndexEntry(@Nonnull Key.Evaluated key, @Nonnull Key.Evaluated value) {
         }
     }
 
-    public IndexEntry(@Nonnull Key.Evaluated key) {
-        this(key, Key.Evaluated.EMPTY);
+    @API(API.Status.INTERNAL)
+    public IndexEntry(@Nonnull Index index, @Nonnull Key.Evaluated key) {
+        this(index, key, Key.Evaluated.EMPTY);
     }
 
-    public IndexEntry(@Nonnull Tuple key, @Nonnull Tuple value) {
+    @API(API.Status.INTERNAL)
+    public IndexEntry(@Nonnull Index index, @Nonnull Tuple key, @Nonnull Tuple value) {
         this.key = key;
         this.value = value;
+        this.index = index;
```

Comment:
This is minor and pedantic but it would be nice to keep the assignments int the same order as the parameters (so `this.index = index` goes at the beginning).

Reply:
Woops, yeah.